### PR TITLE
change type to double for derived vars in cprnc

### PR DESCRIPTION
**Pull request by jedwards4b**
Change type of derived variables to double - consider this in addition to 3530

Test suite: 
Test baseline: 
Test namelist changes: 
Test status: bit for bit

Fixes 3525

User interface changes?: 

Update gh-pages html (Y/N)?:

Code review:

### CODE REVIEW

**Comment by jedwards4b** [comment]
@billsacks thanks for letting me know about the tests, I'll fix this.

**Comment by jedwards4b** [comment]
This test now passes, however I'm sure there are more formatting problems - can we add a make check to the Makefile to run testing for cprnc?  I'm happy to add this target.

**Comment by jedwards4b** [comment]
@billsacks all tests now pass and I've added a make check target to the Makefile

File: config/cesm/machines/config_machines.xml [hunk 1]
```diff
@@ -369,7 +369,8 @@ This allows using a different mpirun command to launch unit tests
     <DIN_LOC_ROOT_CLMFORC>/glade/p/cgd/tss/CTSM_datm_forcing_data</DIN_LOC_ROOT_CLMFORC>
     <DOUT_S_ROOT>$CIME_OUTPUT_ROOT/archive/$CASE</DOUT_S_ROOT>
     <BASELINE_ROOT>$ENV{CESMDATAROOT}/cesm_baselines</BASELINE_ROOT>
-    <CCSM_CPRNC>$ENV{CESMDATAROOT}/tools/cime/tools/cprnc/cprnc.cheyenne</CCSM_CPRNC>
+<!--    <CCSM_CPRNC>$ENV{CESMDATAROOT}/tools/cime/tools/cprnc/cprnc.cheyenne</CCSM_CPRNC> -->
+    <CCSM_CPRNC>/glade/work/jedwards/sandboxes/cesm2_x_alpha/cime/tools/cprnc/cprnc</CCSM_CPRNC>
```
Comment (billsacks) [inline config/cesm/machines/config_machines.xml:373]: - [ ] Can you please revert this change? (I assume it was a temporary thing?)

**Comment by jedwards4b** [comment]
@billsacks please feel free to push additional changes.  I've reverted the change to config_machines.xml

**Comment by billsacks** [comment]
- [ ] @jedwards4b There is just one test with problematic output in one field: `cprnc test_inputs/clm2.h1.subset.control.nc test_inputs/clm2.h1.subset.test.nc`.

On master, the output for `land1d_ityplunit` looks like:

```
 land1d_ityplunit   (landunit)
        701      935  (   132) (    12) (    13) 
                 935        9        1       6       2       0       0
                 935        8        1               8               0
                 935  (    13) (    12)
 RMS land1d_ityplunit                         3            NORMALIZED          1
```

whereas on your branch it looks like:

```
 land1d_ityplunit   (landunit)
        701      935  (   132) (    12) (    13) 
                 935   9.000000000000000E+00   10000.00000000000E-04
              60000.00000000000E-04   00000.00000000000E+00     935          80000.00000000000E-04           10000.00000000000E-04
                   8   0.000000000000000E+00
                 935  (    13) (    12)
 RMS land1d_ityplunit                 3.3332E+00            NORMALIZED  7.8800E-01
```

Note the extra line break and the garbage characters in the new version. I'm pretty sure the problem is in the 903 format, but I haven't gotten further than that. Can you please fix that when you get a chance?

**Comment by jedwards4b** [comment]
I don't see that formatting issue, but try again now.

**Comment by billsacks** [comment]
It's better, but still incorrect.

Old on left `<`, new on right`>`:

```diff
/Users/sacks/temporary/cime/tools/cprnc/testout/clm2.h1.subset.test.nc.out
51,52c51,52
<                  935        9        1       6       2       0       0
<                  935        8        1               8               0
---
>                  935   0.900000000000000E+01  0.100000000000000E+01  0.600000000000000E+01       2  0.000000000000000E+00
>               0.000000000000000E+00      935  0.800000000000000E+01   0.100000000000000E+01       8  0.000000000000000E+00
```

Looking quickly at the difference in formats, old vs. new, it looks like there are indeed a different number of numeric values on the first vs. second line in the old vs. new:

Old: 7 integers before the new line

```Fortran
903   format(12x,      i8,1x,i8,a,5i8,/, &
             12x,      i8,1x,  i8,a,i8,8x,  i8,8x,  i8)
```

New: 1 integer + 3 floats + 1 integer + 1 float = 6 numeric values before the new line:

```Fortran
903   format(12x,      i8,1x,3e23.15,i8,e23.15/, &
             12x,      e23.15,1x,i8,e23.15,a,e23.15,i8,e23.15)
```

But I haven't looked closely enough to know what the correct fix is.

**Comment by jedwards4b** [comment]
how about now
